ci/unit/test_main_node.sh

```diff
@@ -22,16 +22,16 @@ sudo ln -sf "$SL_PREFIX/share/spdk/scripts/setup.sh" /usr/share/spdk/scripts
 sudo ln -sf "$SL_PREFIX/share/spdk/scripts/common.sh" /usr/share/spdk/scripts
 sudo ln -s "$SL_PREFIX/include"  /usr/share/spdk/include
 
-if [ "$WITH_VALGRIND" = "disabled"  ]; then
-    # set CMOCKA envs here
-    export CMOCKA_MESSAGE_OUTPUT=xml
-    export CMOCKA_XML_FILE="$DAOS_BASE"/test_results/%g.xml
-fi
+# set CMOCKA envs here
+export CMOCKA_MESSAGE_OUTPUT=xml
+export CMOCKA_XML_FILE="${DAOS_BASE}/test_results/%g"
+[[ -z ${WITH_VALGRIND} ]] && export "${CMOCKA_XML_FILE}.xml" \
+    || export "${CMOCKA_XML_FILE}_${WITH_VALGRIND}.xml"
```

Comment:
@marcelarosalesj Did you note the shellcheck advise here?

```suggestion
if [ -z ${WITH_VALGRIND} ]; then
    export "${CMOCKA_XML_FILE}.xml"
else
    export "${CMOCKA_XML_FILE}_${WITH_VALGRIND}.xml"
```
is the correct syntax.

But again, I would question why this distinction needs to be here.  The name of the file can be the same no matter if valgrind is being used or not.  It's only in the very final stages of archiving artifacts that you need to change the name of the directory/files so they are unique as artifacts.

Reply:
The cmocka xmls are going to be publish by junit for both UT and UT with memcheck, we need a way to distinguish both in "Test results". This is because if the memcheck stage has errors, we can go to its cmocka tests results for reference.